tests/integration/test_decompositions_integration.py

```diff
@@ -122,7 +122,7 @@ def test_graph_embed_identity(self, setup_eng, tol):
         """Test that nothing is done if the adjacency matrix is the identity"""
         prog = sf.Program(3)
         with prog.context as q:
-            ops.GraphEmbed(np.identity(6)) | q
```

Comment:
purely for speed?

Reply:
No, the Decomposition infers the number of modes it acts on from the size of the matrix. 6 was wrong, but it did not cause an error because of another bug in the code.

Reply:
Or more specifically the matrix size must match the number of modes it acts on.